src/raftstore/store/fsm/apply.rs

```diff
@@ -347,15 +351,21 @@ impl ApplyContext {
             self.kv_wb_last_keys = 0;
         }
         self.cbs.push(ApplyCallback::new(delegate.region.clone()));
-        self.last_applied_index = delegate.apply_state.get_applied_index();
+        self.cur_region_id = delegate.region_id();
+        self.last_applied_index
+            .insert(self.cur_region_id, delegate.apply_state.get_applied_index());
```

Comment:
em, can it hurt the performance?

Reply:
It's a little heavy than previous, but the hash map won't be very large (it's cleared for every batch). So I guess it's ok.